File: sdk/common/src/main/java/ink/reactor/sdk/config/ConfigService.java
package ink.reactor.sdk.config;

import java.io.IOException;
import java.nio.file.Path;
import java.util.Collection;

public interface ConfigService {
    ConfigSection load(final Path path);
    void save(final Path path, final ConfigSection section, final SaveOptions options);

    ConfigSection createIfAbsentAndLoad(final String fileName, final Path outDestination, final ClassLoader classLoader) throws IOException;
    default ConfigSection createIfAbsentAndLoad(final String fileName, final ClassLoader classLoader) throws IOException {
        return createIfAbsentAndLoad(fileName, Path.of(fileName), classLoader);
    }

    Collection<String> fileExtensions();
}

File: sdk/common/src/main/kotlin/ink/reactor/sdk/config/ConfigService.kt
package ink.reactor.sdk.config

import java.io.IOException
import java.nio.file.Path

interface ConfigService {
    val fileExtensions: List<String>

    fun load(path: Path): ConfigSection
    fun save(path: Path, section: ConfigSection, options: SaveOptions = SaveOptions())

    @Throws(IOException::class)
    fun createIfAbsentAndLoad(fileName: String, outDestination: Path, classLoader: ClassLoader): ConfigSection

    @Throws(IOException::class)
    fun createIfAbsentAndLoad(fileName: String, classLoader: ClassLoader): ConfigSection {
        return createIfAbsentAndLoad(fileName, Path.of(fileName), classLoader)
    }
}
